docs/articles/documentation/recipes/best-practices/best-practices.md

```diff
@@ -0,0 +1,223 @@
+---
+layout: docs
+title: Best Practices
+permalink: /documentation/recipes/best-practices/best-practices.html
+---
+# Best Practices
+
+This article covers the recommended ways to test with TestCafe and covers the following topics:
+
+* Smart Assertions
+* Use of Page Objects
+* Use of Roles for Login
+* File structure
+* Clean up between Tests (beforeEach)
+* E2E Tests Limitations
+* Selectors strategy
+
+## Smart Assertions
+
+In end-to-end web testing, peripheral factors (like network lag, processor, or memory bottlenecks in containers) can interfere with the assertions and render tests inconclusive (otherwise known as 'flaky').
```

Comment:
I don't understand "inconclusive", just never saw it meaning "inconsistent". So, you can ask Dirk if you are sure it's good here. Otherwise, I wouldn't use it.

Reply:
An inconclusive test is one that doesn't tell you anything even if it's green. [Source](https://www.ibm.com/support/knowledgecenter/SSYMRC_7.0.1/com.ibm.rational.test.qm.doc/topics/c_execution_verdicts.html) (search for 'inconclusive' in the linked doc)

It's more of a technical term, but I'll ask Dirk anyway.

Reply:
> In end-to-end web testing, unpredictable factors (like network lag, processor, or memory bottlenecks in containers) can interfere with the assertions and produce inconsistent test results. Such tests are inconclusive (sometimes called 'flaky').